### PR TITLE
Fix Spacepod bullshit.

### LINKED ISSUE #2412: Space Pods Occupant Error
We seem to have a rampant error with Space Pods where if someone gets in them, and then precedes to get out, they get 'stuck' in the pod. They do exit it, and can move about, but the 'occupant' variable gets stuck with them in it. There are two potential causes to this error: 1) when someone exits the pod, it does not change to variable 'occupant = null'; 2) when someone switches to the passenger seat, variable 'occupant2', the variable 'occupant' does not revert to 'occupant = null'.

As a note, included bugs with this do include anyone still registered to variable 'occupant' are still affected by the pod. If the pod were to self-destruct or be destroyed, then whoever is still registered to 'occupant', regardless of where they are, will be teleported back to the pod location (as it ejects them to the pod), and gibbed (as the pod has now exploded). This has happened to someone as they had accidentally left the pod blocking a door against cultists. When they blew it up (as there was no other way to move it), the person who was still registered to 'occupant' received a message saying the pod was critical and had 10 seconds to evacuate. It then proceeded to explode, and teleported him back to it, gibbing him, even though he was on station, and the pod was at telecomms in the medical shuttle.

### PR DESCRIPTION
**Pull request by tigercat2000**
Fixes #2412 as best as I can really.

All references _should_ behave properly, but if they don't, there is a proc now which fixes shit and debug logs it.

### CODE REVIEW

File: code/game/vehicles/spacepods/spacepod.dm [hunk 1]
```diff
@@ -464,129 +468,119 @@
...
+			else
+				user << "<span class='notice'>[occupant] was faster. Try better next time, loser.</span>"
+		else
+			user << "<span class='notice'>You stop entering \the [src].</span>"
 	else
-		user << "You stop entering the spacepod."
+		user << "<span class='danger'>You can't fit in \the [src], it's full!</span>"
+
+/obj/spacepod/proc
+	__SANITY_CHECK_FUCKING_OCCUPANT_VARS()	// /obj/spacepod/proc/__SANITY_CHECK_FUCKING_OCCUPANT_VARS()
```
Comment (tigercat2000) [inline code/game/vehicles/spacepods/spacepod.dm]: It really fucking bothered me to have this actually pathed correctly, so I added a comment for searching.

Comment (Fox-McCloud) [inline code/game/vehicles/spacepods/spacepod.dm]: Reallly think this needs a more sane, less vulgar, and less ragy proc name.

File: code/game/vehicles/spacepods/spacepod.dm [hunk 2]
```diff
@@ -464,129 +468,119 @@
...
+		if(occupant)
+			if(!ismob(occupant))
+				occupant = null
+				occupant.forceMove(get_turf(src))
```
Comment (Fox-McCloud) [inline code/game/vehicles/spacepods/spacepod.dm]: shouldn't occupant be nulled second? What this effectively null out the reference then end up not moving someone to the turf?

Comment (tigercat2000) [inline code/game/vehicles/spacepods/spacepod.dm]: i shouldn't code when annoyed

**Comment by tigercat2000** [comment]
I'll fix this in a little while, busy with something else atm.

**Comment by TheDZD** [comment]
Relative pathing procs is bad, starting procs with underscores is weird, why are you doing both of these?

**Comment by tigercat2000** [comment]
because it's fancy

**Comment by TheDZD** [comment]
That's not really a valid reason for poorly-formatted code.

**Comment by Fox-McCloud** [comment]
Going to have to agree with @TheDZD on this one.

**Comment by tigercat2000** [comment]
whatever

**Pull request merged**